server/issue.go

```diff
@@ -1006,3 +1008,109 @@ func (s *Server) postInboxIssueActivity(ctx context.Context, issue *api.Issue, a
+	deployConfig, err := s.DeploymentConfigService.FindDeploymentConfig(ctx, &api.DeploymentConfigFind{
+		ProjectID: &projectID,
+	})
+	if err != nil {
+		return nil, echo.NewHTTPError(http.StatusInternalServerError, fmt.Sprintf("Failed to fetch deployment config for project ID: %v", projectID)).SetInternal(err)
+	}
+	if deployConfig == nil {
+		return nil, echo.NewHTTPError(http.StatusBadRequest, fmt.Sprintf("Deployment config missing for project ID: %v", projectID)).SetInternal(err)
+	}
+	deploySchedule, err := api.ValidateAndGetDeploymentSchedule(deployConfig.Payload)
+	if err != nil {
+		return nil, echo.NewHTTPError(http.StatusInternalServerError, "Failed to get deployment schedule").SetInternal(err)
+	}
+	for _, database := range databaseList {
+		if err := s.composeDatabaseRelationship(ctx, database); err != nil {
+			return nil, echo.NewHTTPError(http.StatusInternalServerError, fmt.Sprintf("Failed to compose database relationship for database ID %v", database.ID)).SetInternal(err)
+		}
+	}
+	p, err := getPipelineFromDeploymentSchedule(deploySchedule, databaseName, databaseList)
+	if err != nil {
+		return nil, echo.NewHTTPError(http.StatusInternalServerError, "Failed to create deployment pipeline").SetInternal(err)
+	}
+	return p, nil
+}
+
+// getSchemaFromPeerTenantDatabase gets the schema version and schema from peer a peer tenant database.
+// It's used for creating a database in a tenant mode project.
+// When a peer tenant database doesn't exist, we will return an error if there are databases in the project with the same name.
+// Otherwise, we will create a blank database without schema.
+func (s *Server) getSchemaFromPeerTenantDatabase(ctx context.Context, instance *api.Instance, projectID int, databaseName string) (string, string, error) {
+	// Find all databases in the project.
+	databaseList, err := s.DatabaseService.FindDatabaseList(ctx, &api.DatabaseFind{
+		ProjectID: &projectID,
+	})
+	if err != nil {
+		return "", "", echo.NewHTTPError(http.StatusInternalServerError, fmt.Sprintf("Failed to fetch databases in project ID: %v", projectID)).SetInternal(err)
+	}
+
+	pipeline, err := s.getTenantDatabases(ctx, projectID, databaseList, databaseName)
+	if err != nil {
+		return "", "", err
+	}
+	similarDB := getPeerTenantDatabase(pipeline, instance.EnvironmentID)
+
+	// When there is no existing tenant, we will look at all existing databases in the tenant mode project.
+	// If there are existing databases with the same name, we will disallow the database creation.
+	// Otherwise, we will create a blank new database.
+	if similarDB == nil {
+		found := false
+		for _, db := range databaseList {
+			if db.Name == databaseName {
+				found = true
+				break
+			}
+		}
+		if found {
+			err := fmt.Errorf("Peer tenant database isn't found, but there are existing databases with the same name %q; the request is not allowed because of ambiguity", databaseName)
```

Comment:
Conflicting database name, project has existing database named %q, but it's not from the selected peer tenants.